src/lib.rs

```diff
@@ -1080,6 +1080,14 @@ impl Graphics2D
             vertex_circle_coords_normalized
         );
     }
+
+    /// Sets the current clip to the rectangle specified by the given
+    /// coordinates. Rendering operations have no effect outside of the
+    /// clipping area.
+    pub fn set_clip(&mut self, x: i32, y: i32, width: i32, height: i32)
```

Comment:
Taking a `Rectangle` here should make it nice and easy to get the coordinates in the form OpenGL wants. It has functions for `width()` and `height()`, and also lets you get the `bottom_left()` coordinates :)

Reply:
Is there really the need for user to construct `Rectangle` and for lib to destructure it further? I think it is unnecessary overhead :-/

Reply:
Hmm I see what you're saying, however it's good to keep things consistent with existing functions like `draw_rectangle()`. While in some cases it's less convenient for the user to have to build the struct, in other cases it's more convenient, as the user is probably already using `Vector2` structs to represent the position/size of the area.

Reply:
(also performance should be identical -- compilers normally break up structs into individual values in these situations. There may be a couple of extra additions/subtractions to find the bottom right coord, but these will be unnoticeable compared to the draw call itself)